numba/misc/numba_gdbinfo.py

```diff
@@ -1,84 +1,143 @@
 """Module for displaying information about Numba's gdb set up"""
+from collections import namedtuple
+import os
+import re
+import shutil
+import subprocess
+from textwrap import dedent
+from numba import config
 
+# Container for the output of the gdb info data collection
+_fields = ('binary_loc, extension_loc, py_ver, np_ver, supported')
+_gdb_info = namedtuple('_gdb_info', _fields)
 
-def _run_cmd(cmdline, env):
-    """Runs cmdline (list of string args) in a subprocess with environment dict
-    env.
-    """
-    import subprocess
-    import threading
-    popen = subprocess.Popen(cmdline,
-                             stdout=subprocess.PIPE,
-                             stderr=subprocess.PIPE,
-                             env=env)
-    # finish in 10 seconds or kill it
-    timeout = threading.Timer(10, popen.kill)
-    try:
-        timeout.start()
-        out, err = popen.communicate()
-        if popen.returncode != 0:
-            raise AssertionError(
-                "process failed with code %s: stderr follows\n%s\n" %
-                (popen.returncode, err.decode()))
-        return out.decode(), err.decode()
-    finally:
-        timeout.cancel()
-    return None, None
 
+class _GDBTestWrapper():
+    """Wraps the gdb binary and has methods for checking what the gdb binary
+    has support for (Python and NumPy)."""
 
-def display_gdbinfo(sep_pos=45):
+    def __init__(self,):
+        gdb_binary = shutil.which(config.GDB_BINARY)
+        if gdb_binary is None:
+            msg = ("No valid binary could be found for gdb named: "
+                   f"{config.GDB_BINARY}")
+            raise ValueError(msg)
```

Comment:
with a  invalid path for gdb, this exception will trigger the following error message:
```bash
$ NUMBA_GDB_BINARY=jifoj numba -g
GDB info:
Traceback (most recent call last):
  File "/root/miniconda3/envs/numba_dev/bin/numba", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/root/dev/numba/bin/numba", line 8, in <module>
    main()
  File "/root/dev/numba/numba/misc/numba_entry.py", line 54, in main
    display_gdbinfo()
  File "/root/dev/numba/numba/misc/numba_gdbinfo.py", line 127, in display_gdbinfo
    gdb_info = collect_gdbinfo()
  File "/root/dev/numba/numba/misc/numba_gdbinfo.py", line 70, in collect_gdbinfo
    gdb_wrapper = _GDBTestWrapper()
  File "/root/dev/numba/numba/misc/numba_gdbinfo.py", line 24, in __init__
    raise ValueError(msg)
ValueError: No valid binary could be found for gdb named: jifoj
```

Reply:
That was intended, if the `gdb` binary is invalid the rest of the information from `numba -g` is empty/irrelevant? Were you hoping something else might happen, perhaps to just report that the binary is invalid as part of the output?

Reply:
Recorded that this needs some attention here: https://github.com/numba/numba/issues/7792